firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Ranges.txt

```diff
@@ -0,0 +1,100 @@
+Scenario:The Ranges:  clamp and within function works correctly
+  GIVEN('The clamp and within function')
+//floor>value>ceiling
+  WHEN('the function clamp is called with a floor of 1, a ceiling of 255, and a test value of 131')
+    THEN('Returns correct value 131') 
+
+  WHEN('floor is '1 and ceiling is 255 , value 131 is within range')
+    THEN('Returns true ,  value 131 is in range')
+
+
+//value >ceiling> floor 
+   WHEN('The function clamp is called with a floor of 1, a ceiling of 10, and a test value of 131')
+     THEN('Returns expected value 10')
+
+   WHEN('The function within is called with a floor of 1, a ceiling of 10, and a test value of 131')  
+      THEN('Returns false, value is not in range')
+
+
+//value > floor > ceiling
+  GIVEN('The clamp and within function')
+  WHEN('The function clamp is called with a floor of 10, a ceiling of 131, and a test value of 8')
+    THEN('Returns expected floor value 10')
+
+  WHEN('The function clamp is called with a floor of 10, a ceiling of 131, and a test value of 8')  
+     THEN('Returns false, value is not in range')
+
+//floor = value > ceiling
+  WHEN('The function clamp is called with a floor of 8, a ceiling of 131, and a test value of 8')
+    THEN('Returns expected  input value 8') 
+
+  WHEN('The function clamp is called with a floor of 8, a ceiling of 131, and a test value of 8')   
+     THEN('Returns true, value is in range')
+
+//floor > ceiling > value 
+  GIVEN('The clamp and within function')  
+  WHEN('The function clamp is called with a floor of 8, a ceiling of 255, and a test value of 83')
+    THEN('Returns expected  input value 83')
+
+  WHEN('The function within is called with a floor of 8, a ceiling of 255, and a test value of 83')  
+    THEN('Returns true, value is in range')
+
+//floor = ceiling = value
```

Comment:
For example, this should clearly be `floor = ceiling < value`. Once these comments are corrected, then it'll be easier for me to confirm coverage of https://github.com/pez-globo/pufferfish-software/pull/366#discussion_r654132679 .